File: RedSheet.DbEntities/Models/SuggestedAction.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("SuggestedActions", Schema = "dbo")]
	[Serializable]
	public partial class SuggestedAction  
	{
 
	[Range(1, int.MaxValue)]
	public int CultureCountryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int SuggestedActionId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string SuggestedActionValue { get; set; } 

	public SuggestedAction()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/suggested-action.ts
import { required, maxLength, range } from '@rx/annotations';

export class SuggestedAction {
    constructor(suggestedAction?: SuggestedAction )  {
        let properties = [ "cultureCountryId", "suggestedActionId", "suggestedActionValue",];
        for (let property of properties)
            if (suggestedAction && suggestedAction[property])
                this[property] = suggestedAction[property];
    }
 
    @range(1,2147483647)
	cultureCountryId : number =   undefined;
 
	suggestedActionId : number =   0 ;
 
    @required()
    @maxLength(500)
	suggestedActionValue : string =   undefined;
}
